constructor_io/constructor_io.py

```diff
@@ -45,24 +66,44 @@ def _make_url(self, endpoint, params=None):
         return "{0}://{1}/{2}?{3}".format(self._protocol, self._host, endpoint,
                                           self._serialize_params(params))
 
-    def query(self, query_str):
+    def __make_server_request(self, request_method, *args, **kwargs):
+        retries_left = self._server_error_retries
+
+        while True:
+            try:
+                # Wrap server error codes as exceptions
+                response = request_method(*args, **kwargs)
+                if round(response.status_code / 100) == 5:
+                    raise ConstructorServerError(response.text)
+                elif round(response.status_code / 100) == 4:
+                    raise ConstructorInputError(response.text)
+                elif not round(response.status_code / 100) == 2:
+                    raise ConstructorError(response.text)
+                return response
+            except ConstructorServerError as error:
+                # Retry in case of server error
+                if retries_left <= 0:
+                    raise error
+                timeout = RETRY_TIMEOUT_IN_CASE_OF_SERVER_ERROR * (
```

Comment:
This seems to increment the timemout linearly, while usually timeouts increment exponentially,
so I'd recommend something like:
```
timeout = <default>

try:
...
except <exception>:
     time.sleep(timeout)
     timeout **= 2
```

Reply:
If you still prefer to increment the timeout linearly, a modified version of the code above is more readable in my opinion:
```
timeout = <default>

try:
...
except <exception>:
     time.sleep(timeout)
     timeout += <default>
```